ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto

```diff
@@ -128,6 +135,57 @@ message GetFlatTransactionResponse {
   Transaction transaction = 1;
 }
 
+message GetEventsByContractIdRequest {
+  // The contract id being queried
+  // Required
+  string contract_id = 1;
+
+  // The parties whose events the client expects to see.
+  // Events that are not visible for the parties in this collection will not be present in the response.
+  // Each element be a valid PartyIdString (as describe in ``value.proto``).
+  // Required
+  repeated string requesting_parties = 2;
+
+}
+message GetEventsByContractIdResponse {
+  // The collection of events, will only contain consumed ExerciseEvents
+  repeated TreeEvent events = 1;
+}
+
+message GetEventsByContractKeyRequest {
+
+  // The contract key to search for.
+  // Required
+  Value contract_key = 1;
+
+  // The parties whose events the client expects to see.
+  // Events that are not visible for the parties in this collection will not be present in the response.
+  // Each element be a valid PartyIdString (as describe in ``value.proto``).
+  // Required
+  repeated string requesting_parties = 2;
+
+  // Optional - will default to 1000
+  int32 max_responses = 3;
```

Comment:
```suggestion
  int32 max_events = 3;
```
I take this value controls the number of events returned in the response.
If there is more matching results then which events will be included in the response? The newest ones?